src/DynamoCore/Core/Threading/UpdateGraphAsyncTask.cs

```diff
@@ -103,37 +120,37 @@ protected override void HandleTaskCompletionCore()
             }
         }
 
+        /// <summary>
+        /// Return if this task's graph sync data is a super set of the other
+        /// </summary>
+        /// <param name="other"></param>
+        /// <returns></returns>
+        private bool Contains(UpdateGraphAsyncTask other)
+        {
+            // Be more conservative here. Not only check ModifiedNodes, but
+            // also check *all* nodes in graph sync data. For example, consider
+            // a CBN outputs to a node, the node is a downstream node of cbn.
+            //
+            // Now if node is modified and request a run, task's ModifiedNodes
+            // will cotain this node; then CBN is modified and request a run,
+            // ModifiedNodes would contain both CBN and the node, and previous
+            // task will be thrown away.
+            return other.graphSyncData.NodeIDs.All(graphSyncData.NodeIDs.Contains) && 
+                   other.ModifiedNodes.All(ModifiedNodes.Contains); 
+        }
```

Comment:
No. In this case, 
* foo's `UpdateGraphAysncTask: { ModifiedNodes: {foo}, NodeIDs: {foo}}`
* cbn's `UpdateGraphAsyncTask: {ModifiedNodes: {cbn, foo}, NodeIDs: {cbn}}`.

Notice these two tasks have different `NodeIDs` set. As `NodeIDs` is the set of nodes that finally be executed, it is more accurate than `ModifiedNodes`.